### ISSUE TITLE
hare: should not output empty block ID when hare fails

### ISSUE THREAD

**Issue by countvonzero** [spacemeshos/go-spacemesh#3446]
## Description

when hare fails locally, it should not output empty block ID. instead, it should either
- wait for the sync flow to sync the block certificate from its peers
- wait out the zdist layers. after zdist layer has passed, hare output no longer matter for that layer because tortoise will use global opinions for that layer.

this is related to https://github.com/spacemeshos/go-spacemesh/issues/3445.
in that case, hare output empty block ID. TestScheduleBasic/verify fails because it happens to use this node as the reference hash, and all other nodes have the same hash.
 
![Screenshot from 2022-08-14 16-17-41](https://user-images.githubusercontent.com/30611210/184555265-0dc3feee-0b37-44aa-a046-9a79bab0f92e.png)

**Comment by countvonzero** [comment]
a fix was applied in https://github.com/spacemeshos/go-spacemesh/pull/3443 that caused TestFailedNodes to fail.

https://github.com/spacemeshos/go-spacemesh/compare/9cdf2d13286d4a77e39ab29e9a7165e4493482d5..60c1631226723c0be88f6252fac3b5a59662d62e

the test probably need more time to recover from a cluster-wide hare failure.

**Comment by countvonzero** [comment]
TestFailedNodes:
when testing the behavior that hare does NOT output empty layer upon failure.
the whole cluster's verified layer stuck at layer 14. 

hare failed on layer 15. but turned out this is a network-wide failure and layer 15 indeed is empty. 
a node then could not create vote diff against abstaining on layer 15 because there is no blocks to vote against in layer 15.

from @dshulyak 

there is definitely a problem with terminating empty layers. in that env layers through 15-19 are empty
so ballots encode abstain votes for them as expected, after some distance they drop abstain and should encode against
nodes select ballots with abstain as a base ballot, and against should be in the diff
but it doesn't make sense because there is no block to be encoded in against
so basically they are forever abstain on layer 15 because of the recursive vote decoding

**Comment by dshulyak** [comment]
I will work on revisiting abstain votes, i think there is some undefined behavior with them.
